generic-ec/src/arithmetic.rs

```diff
@@ -432,13 +477,42 @@ impl_unary_ops! {
     Neg (neg Scalar<E>) scalar::neg,
     Neg (neg NonZero<Point<E>>) laws::neg_nonzero_point_is_nonzero_point,
     Neg (neg NonZero<Scalar<E>>) scalar::neg_nonzero,
+    Neg (neg NonZero<SecretScalar<E>>) scalar::neg_nonzero_secret,
 }
 
 impl_op_assign! {
     Point<E>, AddAssign, Point<E>, add_assign, +,
+    Point<E>, AddAssign, NonZero<Point<E>>, add_assign, +,
+    Point<E>, AddAssign, Generator<E>, add_assign, +,
+
+    Point<E>, SubAssign, Point<E>, sub_assign, +,
+    Point<E>, SubAssign, NonZero<Point<E>>, sub_assign, +,
+    Point<E>, SubAssign, Generator<E>, sub_assign, +,
+
     Point<E>, MulAssign, Scalar<E>, mul_assign, *,
+    Point<E>, MulAssign, NonZero<Scalar<E>>, mul_assign, *,
+    Point<E>, MulAssign, SecretScalar<E>, mul_assign, *,
+    Point<E>, MulAssign, NonZero<SecretScalar<E>>, mul_assign, *,
+
     Scalar<E>, AddAssign, Scalar<E>, add_assign, +,
+    Scalar<E>, AddAssign, NonZero<Scalar<E>>, add_assign, +,
+    Scalar<E>, AddAssign, SecretScalar<E>, add_assign, +,
+    Scalar<E>, AddAssign, NonZero<SecretScalar<E>>, add_assign, +,
+
+    Scalar<E>, SubAssign, Scalar<E>, sub_assign, +,
+    Scalar<E>, SubAssign, NonZero<Scalar<E>>, sub_assign, +,
+    Scalar<E>, SubAssign, SecretScalar<E>, sub_assign, +,
+    Scalar<E>, SubAssign, NonZero<SecretScalar<E>>, sub_assign, +,
+
     Scalar<E>, MulAssign, Scalar<E>, mul_assign, *,
+    Scalar<E>, MulAssign, NonZero<Scalar<E>>, mul_assign, *,
+    Scalar<E>, MulAssign, SecretScalar<E>, mul_assign, *,
+    Scalar<E>, MulAssign, NonZero<SecretScalar<E>>, mul_assign, *,
+
+    NonZero<Point<E>>, MulAssign, NonZero<Scalar<E>>, mul_assign, *,
+    NonZero<Point<E>>, MulAssign, NonZero<SecretScalar<E>>, mul_assign, *,
+    NonZero<Scalar<E>>, MulAssign, NonZero<Scalar<E>>, mul_assign, *,
+    NonZero<Scalar<E>>, MulAssign, NonZero<SecretScalar<E>>, mul_assign, *,
```

Comment:
Missing NonZero<Point> by generator.

Also missing a bunch of other NonZeros in left hand positions